File: Week03/LeetCode-2037-2.cpp
class Solution {
public:
    int minMovesToSeat(vector<int>& seats, vector<int>& students) {
        #define MAX_NUM 101
        vector<int> S(MAX_NUM, 0);
        vector<int> C(MAX_NUM, 0);

        // Perform counting sort
        for (auto seat: seats) C[seat]++;
        for (auto student: students) S[student]++;
        
        for (int i=0, k=0; i<MAX_NUM; i++)
            while (C[i]-- > 0)
                seats[k++] = i;

        for (int i=0, k=0; i<MAX_NUM; i++)
            while (S[i]-- > 0)
                students[k++] = i;
        
        int result = 0;
        for (int i=0; i<students.size(); i++)
            result += std::abs(students[i]-seats[i]);
        return result;
    } // end-minMovesToSeat
};

File: Week03/LeetCode-2037-2.py
class Solution:
    def minMovesToSeat(self, seats: List[int], students: List[int]) -> int:
        MAX_NUM = 101
        S = [0]*MAX_NUM
        C = [0]*MAX_NUM
 
        #Perform counting sort
        for seat in seats: C[seat] += 1
        for student in students: S[student] += 1
    
        k = 0
        for i in range(MAX_NUM):   
            while (C[i] > 0):
                seats[k] = i
                k += 1
                C[i] -= 1

        k = 0
        for i in range(MAX_NUM):   
            while (S[i] > 0):
                students[k] = i
                k += 1
                S[i] -= 1
                
        result = 0
        for student, seat in zip(students, seats):
            result += abs(student-seat)
        return result